packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -35,19 +35,28 @@ import "./ISignalService.sol";
 contract SignalService is AuthorizableContract, ISignalService {
     using SafeCast for uint256;
 
+    /// @dev Meaning, for this chainId A, signal service contract B, in given snapshot world state
+    /// root 'C', then here is a storage proof for that signal service.
```

Comment:
Dani, I have a different idea of caching the info.

Reply:
Feel free to proceed, however you like, maybe best on a separate PR to see.